### PR TITLE
Apply dictionary renderer to feature layer

### PR DESCRIPTION
**Pull request by prupani-7**
## Description

PR to add a new Kotlin sample _"Apply Dictionary Renderer to Feature Layer"_ in `Visualization` category.

## Links and Data

Sample Epic: `runtime/kotlin/issues/1942`

## What To Review

-  Review the code to make sure it is easy to follow like other samples on Android
- `README.md` and `README.metadata.json` files

### CODE REVIEW

File: apply-dictionary-renderer-to-feature-layer/README.metadata.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    "relevant_apis": [
+        "DictionaryRenderer",
+        "DictionarySymbolStyle",
+        "FeatureLayer",
+        "Geodatabase"
+    ],
+    "snippets": [
+        "src/main/java/com/esri/arcgismaps/sample/applydictionaryrenderertofeaturelayer/DownloadActivity.kt",
+        "src/main/java/com/esri/arcgismaps/sample/applydictionaryrenderertofeaturelayer/MainActivity.kt"
+    ],
```
Comment (shubham7109) [inline apply-dictionary-renderer-to-feature-layer/README.metadata.json:28]: I realize this is just an annoying quirk of the script but `MainAcitivty.kt` should be set as the first snippet.

File: apply-dictionary-renderer-to-feature-layer/src/main/java/com/esri/arcgismaps/sample/applydictionaryrenderertofeaturelayer/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            dictionarySymbolStyle.load().getOrElse {
+                return@launch showError("Error loading DictionarySymbolStyle: ${it.message}")
+            }
+        }
+
+        lifecycleScope.launch {
+            // load the geodatabase
+            geodatabase.load().getOrElse {
+                showError("Error loading Geodatabase: ${it.message}")
+            }
```
Comment (shubham7109) [inline apply-dictionary-renderer-to-feature-layer/src/main/java/com/esri/arcgismaps/sample/applydictionaryrenderertofeaturelayer/MainActivity.kt:88]:
```suggestion
        lifecycleScope.launch {
            // load the dictionary symbol style
            dictionarySymbolStyle.load().getOrElse {
                return@launch showError("Error loading DictionarySymbolStyle: ${it.message}")
            }

            // load the geodatabase
            geodatabase.load().getOrElse {
                showError("Error loading Geodatabase: ${it.message}")
            }
```

Loading both `dictionarySymbolStyle` and the `geodatabase` in two different scopes would create a possibility that one would load before the other causing a race condition. Suggest just loading both in one coroutine scope.

**Review by shubham7109 — APPROVED** [review]
Just a few minor comments. Sample works well 👍

File: apply-dictionary-renderer-to-feature-layer/src/main/java/com/esri/arcgismaps/sample/applydictionaryrenderertofeaturelayer/MainActivity.kt (hunk 2 above)
Comment (prupani-7) [reply apply-dictionary-renderer-to-feature-layer/src/main/java/com/esri/arcgismaps/sample/applydictionaryrenderertofeaturelayer/MainActivity.kt]: good catch, applied the suggestion

File: apply-dictionary-renderer-to-feature-layer/src/main/java/com/esri/arcgismaps/sample/applydictionaryrenderertofeaturelayer/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                }
+                val featureLayer = FeatureLayer(geodatabaseFeatureTable)
+                featureLayer.load().getOrElse {
+                    return@launch showError("Error loading FeatureLayer: ${it.message}")
+                }
+                // add featureLayer to the map's operational layer
+                mapView.map?.operationalLayers?.add(featureLayer)
+
+                // create dictionaryRenderer using the dictionarySymbolStyle and apply it to the featureLayer's renderer
+                val dictionaryRenderer = DictionaryRenderer(dictionarySymbolStyle)
```
Comment (hud10837) [inline apply-dictionary-renderer-to-feature-layer/src/main/java/com/esri/arcgismaps/sample/applydictionaryrenderertofeaturelayer/MainActivity.kt:101]: Would it work to create a single DictionaryRenderer and apply it to every feature layer? That's a bit of a nitpick, and I'm not sure the optimization wouldn't just make it harder to read the sample, so I would leave it up to you.

**Review by hud10837 — APPROVED** [review]
One thought, but happy to approve either way

**Comment by prupani-7** [comment]
Thanks @hud10837 @shubham7109 for the reviews! :) 
Merging into v.next

**Pull request merged**